src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1077,6 +1103,15 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 			}
 
 			queryTableCount++;
+
+			/* we do not expect to see a view in modify query */
+			if (rangeTableEntry->relkind == RELKIND_VIEW)
+			{
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("cannot perform distributed planning for the "
```

Comment:
Another note : it is impossible to hit this error anymore. 
```c
	Var *partitionColumn = PartitionColumn(distributedTableId, rangeTableId);
```
In the beginning of the function errors out before hitting this one.